### PR TITLE
no Content-Length in header if response code is 1xx or 204

### PR DESCRIPTION
**Pull request by PrajaktaPurohit**
Do not insert Content-Length in header if response code is 1xx or 204.

REF: https://tools.ietf.org/html/rfc7230#section-3.3.3

### CODE REVIEW

File: src/webmachine_request.erl [hunk 1]
```diff
@@ -724,24 +724,28 @@ make_version({1, 0}) ->
 make_version(_) ->
     <<"HTTP/1.1 ">>.
 
+-spec update_header_with_content_length(number(), atom(), term()) -> number().
+update_header_with_content_length(Code, _Length, RD) when Code >= 100,
```
Comment (seancribbs) [inline src/webmachine_request.erl]: Your guard has superfluous clauses. `when Code == 204; Code == 205; Code == 304` would be sufficient (for responses that do not require `Content-Length`).

**Review by seancribbs — CHANGES_REQUESTED** [review]
Please add a test that demonstrates the bug and that your patch fixes the problem.

File: src/webmachine_request.erl (hunk 1 above)
Comment (seancribbs) [reply src/webmachine_request.erl]: Also, Webmachine does not reply with responses in the `100` to `199` range (e.g. `100 Continue`), those are handled by mochiweb.

**Comment by markan** [comment]
@seancribbs I'd like to try and revive this PR. 

I'm picking up this from someone else, and so missing a little bit of history and context. I see that you requested tests, and I see that there are tests here: https://github.com/webmachine/webmachine/blob/d7475f219bf2e4ae0aa151733e233e5b40f26741/src/webmachine_request.erl#L948, but I'm not sure whether those were added as a response, or were insufficient and needed improvement. Are those tests sufficient?

For cleanliness I've rebased it on master. The travis test failures seem to be the same ones mentioned in #292.

**Comment by seancribbs** [comment]
Thank you for the update, @markan. I have also forgotten what other tests were needed, but the code as given fits with my knowledge of the HTTP spec. Merging.

**Pull request merged**